src/Highway.Data.EntityFramework/Contexts/ReadonlyDataContext.cs

```diff
@@ -0,0 +1,210 @@
+        /// <param name="mapping">The Mapping Configuration that will determine how the tables and objects interact</param>
+        /// <param name="contextConfiguration">The context specific configuration that will change context level behavior</param>
+        /// <param name="log">The logger being supplied for this context ( Optional )</param>
+        public ReadonlyDataContext(
+            DbConnection dbConnection,
+            bool contextOwnsConnection,
+            IMappingConfiguration mapping,
+            IContextConfiguration contextConfiguration,
+            ILog log)
+        {
+            _context = new ReadonlyDbContext(dbConnection, contextOwnsConnection, mapping, contextConfiguration, log);
+        }
+
+        /// <summary>
+        ///     This gives a mockable wrapper around the normal <see cref="DbSet{TEntity}" /> method that allows for testability
+        ///     This method is now virtual to allow for the injection of cross cutting concerns.
+        /// </summary>
+        /// <typeparam name="T">The Entity being queried</typeparam>
+        /// <returns>
+        ///     <see cref="IQueryable{T}" />
+        /// </returns>
+        public IQueryable<T> AsQueryable<T>()
+            where T : class
+        {
+            return _context.InnerSet<T>();
+        }
+
+        public void Dispose()
+        {
+            _context?.Dispose();
+        }
+
+        /// <summary>
+        ///     Executes a SQL command and tries to map the returned dataset into an <see cref="IEnumerable{T}" />
+        ///     The results should have the same column names as the Entity Type has properties
+        /// </summary>
+        /// <typeparam name="T">The Entity Type that the return should be mapped to</typeparam>
+        /// <param name="sql">The Sql Statement</param>
+        /// <param name="dbParams">A List of Database Parameters for the Query</param>
+        /// <returns>An <see cref="IEnumerable{T}" /> from the query return</returns>
+        public IEnumerable<T> ExecuteSqlQuery<T>(string sql, params DbParameter[] dbParams)
+        {
+            return _context.ExecuteSqlQuery<T>(sql, dbParams);
+        }
+
+        /// <summary>
+        ///     Reloads the provided instance of <typeparamref name="T" /> from the database
+        /// </summary>
+        /// <typeparam name="T">The Entity Type being reloaded</typeparam>
+        /// <param name="item">The <typeparamref name="T" /> you want to reload</param>
+        /// <returns>The <typeparamref name="T" /> you reloaded</returns>
+        public virtual T Reload<T>(T item)
+            where T : class
+        {
+            var entry = _context.Entry(item);
+            if (entry == null)
+            {
+                throw new InvalidOperationException("You cannot reload an object that is not in the current Entity Framework data context");
+            }
+
+            entry.Reload();
+
+            return item;
+        }
+    }
```

Comment:
I have added the Reload method.  I don't believe this offers any opportunity to write back to the database, does it?